src/main/java/com/exasol/adapter/dynamodb/documentpath/LoopDocumentPathIteratorState.java

```diff
@@ -0,0 +1,31 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+/**
+ * This class represents the current iteration state of a {@link LoopDocumentPathIterator}.
+ */
+public class LoopDocumentPathIteratorState implements PathIterationStateProvider {
+    private final DocumentPathExpression pathOfThisIterator;
```

Comment:
Can this be a private nested class of LoopDocumentPathIterator? This seems to be it should exist only inside that class.

Reply:
yes